nucypher/network/nodes.py

```diff
@@ -142,19 +149,32 @@ def _signable_interface_info_message(self):
         message = self.canonical_public_address + self.rest_information()[0]
         return message
 
-    def _sign_interface_info(self):
+    def _sign_and_date_interface_info(self):
         message = self._signable_interface_info_message()
-        self._interface_signature_object = self.stamp(message)
+        self._timestamp = maya.now()
+        self._interface_signature_object = self.stamp(self.timestamp_bytes() + message)
 
     @property
     def _interface_signature(self):
         if not self._interface_signature_object:
             try:
-                self._sign_interface_info()
+                self._sign_and_date_interface_info()
             except NoSigningPower:
-                raise NoSigningPower("This Ursula is a Stranger; you didn't init with an interface signature, so you can't verify.")
+                raise NoSigningPower("This Node is a Stranger; you didn't init with an interface signature, so you can't verify.")
         return self._interface_signature_object
 
+    @property
+    def timestamp(self):
+        if not self._timestamp:
+            try:
+                self._sign_and_date_interface_info()
+            except NoSigningPower:
+                raise NoSigningPower("This Node is a Stranger; you didn't init with a timestamp, so you can't verify.")
+        return self._timestamp
+
+    def timestamp_bytes(self):
```

Comment:
Does this really make sense as a method like this? I see this more as a helper/utility type function and not necessarily in this class.

Reply:
Yeah, I wondered the same thing.  But it gets called three times (four if you include the test), so I figure it's at least a good thing to have somewhere.  But yeah, if we use it anywhere else, I think a utility function makes more sense.